freqtrade/freqtradebot.py

```diff
@@ -277,25 +275,28 @@ def process(self) -> None:
             self.strategy.analyze(self.active_pair_whitelist)
 
         with self._exit_lock:
-            # Check for exchange cancellations, timeouts and user requested replace
+            # Check for exchange cancellations, timeouts, and user requested replace
             self.manage_open_orders()
 
-        # Protect from collisions with force_exit.
-        # Without this, freqtrade may try to recreate stoploss_on_exchange orders
-        # while exiting is in process, since telegram messages arrive in an different thread.
         with self._exit_lock:
             trades = Trade.get_open_trades()
-            # First process current opened trades (positions)
-            self.exit_positions(trades)
+            # First process current open trades (positions)
+            closed_trades = self.exit_positions(trades)
+
+        # Attempt to open new trades for pairs that were closed
+        for trade in closed_trades:
+            if self.get_free_open_trades():
+                self.create_trade(trade.pair)
```

Comment:
This will ignore regular pairlist sequence - so this is introducing a bug / unwanted behavior.